docs/code/IDataViewDesignPrinciples.md

```diff
@@ -235,57 +234,91 @@ with a key type.
-provided parameters. Typically, the transform behavior is later serialized.
-When deserialized, the transform is not retrained; its behavior is entirely
-determined by the serialized information.
+ability to compose them to produce efficient data pipelines:
+
+Estimators and transformers. The langauge is derived from a similar idiom in Spark 
+
+A data loader represents a data source as an `IDataView`. A transformer is
+applied via its `Fit` method to an `IDataView` to produce a derived
+`IDataView`. A saver serializes the data produced by an `IDataView` to a
+stream, in some cases in a format that can be read by a loader. There are
+other more specific kinds of components defined and used by the ML.NET code
+base, for example, scorers, evaluators, joins, and caches, but most of the
+aforementioned are internal. While there are several standard kinds of
+components, the set of component kinds is open. In the sequel we discuss the
+most important types of components in the public API of ML.NET.
+
+### Transformers
+
+Transformers are a foundational kind of IDataView component. They have two
+primary responsibilities, from a user's point of view.
+
+As the name suggests, the primary method is `Transform`, which takes
+`IDataView` as input and produce an `IDataView` as output, using the
+`ITransformer.Fit` method. Many transformers simply "add" one or more computed
+columns to their input schema. More precisely, their output schema includes
+all the columns of the input schema, plus some additional columns, whose
+values are computed from some of the input column values. It is common for an
+added column to have the same name as an input column, in which case, the
+added column hides the input column. Both the original column and new column
+are present in the output schema and available for downstream components (in
+particular, savers and diagnostic tools) to inspect. For example, a data view
+that comes from a normalization transformer may, for each slot of a
+vector-valued column named `"Features"`, apply an offset and scale factor and
+bundle the results in a new vector-valued column, also named `"Features"`.
+From the user's perspective (which is entirely based on column names), the
+Features column was "modified" by the transform, but the original values are
+available downstream via the hidden column.
+
+### Estimators
+
+Many transformers require training, meaning that their precise behavior is
+determined from some training data. For example, normalizers and
+dictionary-based mappers, such as the `ValueToKeyMappingTransformer`, build
+their state from training data.
+
+This training occurs with another structure generally parallel to
+`ITransformer` called `IEstimator`, that returns an `ITransformer` once it is
+`Fit`. For example, `NormalizingEstimator` implements
+`IEstimator<NormalizingTransformer>`.
+
+instantiated from user-provided parameters. Typically, the
```

Comment:
`instantiated from user-provided parameters.`

I'm starting to think that @TomFinley didn't write this document. Here we have the _end_ of a sentence with no beginning. However, @TomFinley only _starts_ sentences without finishing them. 😝

Reply:
I did not write this document!

Yes this is a bit of a mess...